scrapers/scrapers/spiders/mo-bills.py

```diff
@@ -0,0 +1,687 @@
+                    )
+                )
+                return
+
+        bill = Bill(
+            identifier=bill_id,
+            title=bill_desc,
+            chamber="lower",
+            legislative_session=self.session,
+            classification=bill_type
+        )
+
+        bill.subject = subs
+        bill.add_title(official_title, note="official")
+        bill_url = f'https://www.house.mo.gov/BillContent.aspx?bill={bid}&year={bill_year}&code={bill_code}&style=new'
+        bill.add_source(bill_url)
+
+        # add sponsors
+        # HEC is a petition for a recount, which can be sponsorless
+        self.parse_house_sponsors(response, bill_id, bill)
+        self.parse_house_actions(response, bill)
+        self.parse_house_bill_versions(response, bill)
+
+        yield bill.as_dict()
+
+    def parse_house_sponsors(self, response, bill_id, bill):
+        bill_sponsors = response.xpath('//BillInformation/Sponsor')
+        for sponsor in bill_sponsors:
+            sponsor_type = sponsor.xpath('./SponsorType/text()').get()
+            if sponsor_type == 'Co-Sponsor':
+                classification = 'co-sponsor'
+            elif sponsor_type == 'Sponsor':
+                classification = 'primary'
+            else:
+                classification = ''
+
+            bill_sponsor = sponsor.xpath('./FullName/text()').get()
+            if bill_sponsor == "" and "HEC" in bill_id:
+                bill.add_sponsorship(
+                    "Petition", entity_type="", classification="primary", primary=True
+                )
+            else:
+                bill.add_sponsorship(
+                    bill_sponsor,
+                    entity_type="person",
+                    classification=classification,
+                    primary=True,
+                )
+
+    def parse_house_actions(self, response, bill):
+        # add actions
+        bill_actions = response.xpath('//BillInformation/Action')
+        old_action_url = ''
+        for action in bill_actions:
+            # new actions are represented by having dates in the first td
+            # otherwise, it's a continuation of the description from the
+            # previous action
+            action_url = action.xpath(
+                './Link/text()').get().replace('.aspx', 'actions.aspx').strip()
+            action_title = action.xpath('./Description/text()').get()
+            action_date = dt.datetime.strptime(
+                action.xpath('./PubDate/text()').get(), '%Y-%m-%d')
+            actor = house_get_actor_from_action(action_title)
+            type_class = get_action(actor, action_title)
+
+            # votes
+            if action.xpath('./RollCall'):
+                rc_yes = action.xpath('./RollCall/TotalYes/text()').get()
+                rc_no = action.xpath('./RollCall/TotalNo/text()').get()
+                rc_present = action.xpath(
+                    './RollCall/TotalPresent/text()').get()
+
+                #     vote = VoteEvent(
```

Comment:
Noticing this is commented out. Did you run into problems with this code? This is primarily a Bills scraper, but there are cases where Vote data is mixed in with Bill data (on the source website), so it's good to yield the VoteEvent items as well.

Reply:
Regarding this issue, I observed that the old scraper produces the VoteEvent items. However, if we choose to yield this vote in Scrapy, it will be in the same file as an item. If you prefer, we can uncomment this part.